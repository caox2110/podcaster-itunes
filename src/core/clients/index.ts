export { default as HttpClientFactory } from './HttpClient.factory';
export * from './enums';
export * from './types';
