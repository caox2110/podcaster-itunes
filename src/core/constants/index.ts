import { appConstant } from './app.constant';
import { themeConstant } from './theme.constant';
import { urlConstant } from './url.constant';
import { queryClientConstant } from './queryClient.constant';

/**
 * Constants for the app
 *
 * @type {{ app: any; theme: any; url: any; }}
 */
const constants = {
  app: appConstant,
  theme: themeConstant,
  url: urlConstant,
  queryClient: queryClientConstant,
};

if (appConstant.IS_DEV) {
  console.warn('Constants are', constants);
  console.warn(`ENV is ${appConstant.MODE}`);
}

export { constants };
