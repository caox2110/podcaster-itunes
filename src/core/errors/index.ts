export { default as ExceptionHandler } from './Exceptions.handler';
