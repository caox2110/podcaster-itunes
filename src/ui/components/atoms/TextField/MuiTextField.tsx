import TextField from '@mui/material/TextField';
import type { TextFieldProps } from '@mui/material';

/**
 * Type of MuiTextField component props.
 * ItÂ´s can receive the MuiTextFieldProps.
 *
 * @typedef {TextFieldProperties}
 */
type TextFieldProperties = {
  id: string;
  name: string;
  placeholder: string;
  'aria-autocomplete'?: string;
  'data-testid'?: string;
} & TextFieldProps;

/**
 * MuiTextField Component definition. An input html element.
 *
 * @export
 * @param {TextFieldProperties} {
  id,
  name,
  variant = 'outlined',
  'aria-autocomplete': ariaAutoComplete,
  autoComplete = 'off',
  type = 'text',
  'data-testid': dataTestId,
  ...rest
}
 * @returns
 */
function MuiTextField({
  id,
  name,
  variant = 'outlined',
  'aria-autocomplete': ariaAutoComplete,
  autoComplete = 'off',
  type = 'text',
  'data-testid': dataTestId,
  ...rest
}: TextFieldProperties) {
  return (
    <TextField
      aria-autocomplete={ariaAutoComplete || 'none'}
      autoComplete={autoComplete}
      id={id}
      inputProps={{ 'data-testid': dataTestId }}
      name={name}
      type={type}
      variant={variant}
      {...rest}
    />
  );
}

/**
 * Default values for MuiTextField component props
 *
 * @type {{ 'aria-autocomplete': string; 'data-testid': string; }}
 */
const defaultProps = {
  'aria-autocomplete': 'none',
  'data-testid': '',
};

MuiTextField.defaultProps = defaultProps;

export type { TextFieldProperties };
export default MuiTextField;
