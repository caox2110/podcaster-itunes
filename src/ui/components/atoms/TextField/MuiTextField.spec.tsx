import { render, screen, userEvent } from '@/core/test';
import type { RenderResult } from '@/core/test';

import MuiTextField, { TextFieldProperties } from './MuiTextField';

type RenderAppType = {
  input: HTMLInputElement;
  label: HTMLLabelElement;
} & RenderResult;

const properties = {
  'data-testid': 'filter-podcast',
  id: 'filter-podcast',
  label: 'Search',
  name: 'filter-podcast',
  placeholder: 'Filter podcasts',
  type: 'text',
};

function renderApp({ ...rest }: TextFieldProperties = properties): RenderAppType {
  const utils = render(<MuiTextField {...rest} />);

  const { getByDataTestId } = utils;

  const input = getByDataTestId(/filter-podcast/) as HTMLInputElement;
  const label = screen
    .getAllByText(/Search/)
    .find((item) => item.id === 'filter-podcast-label') as HTMLLabelElement;

  return {
    ...utils,
    input,
    label,
  };
}

describe('MuiTextField', () => {
  describe('Rendering', () => {
    it('Should render without throwing any errors', () => {
      const { baseElement, container } = renderApp();

      expect(baseElement).toBeTruthy();
      expect(container).toMatchSnapshot();
    });
  });

  describe('Appearance', function () {
    it('should render outlined style by default', () => {
      const { input } = renderApp();
      expect(input.className).toMatch(/outlined/i);
    });
    it('should render autocomplete off by default', () => {
      const { input } = renderApp();
      expect(input.autocomplete).toMatch(/off/i);
    });
    it('should render type search', () => {
      const { input } = renderApp({ ...properties, type: 'search' });
      expect(input.type).toMatch(/search/i);
    });
    it('should render label Search', () => {
      const { label } = renderApp();
      expect(label).toHaveTextContent(/search/i);
    });
    it('should render placeholder Search', () => {
      const { input } = renderApp();
      expect(input.placeholder).toMatch(/filter podcasts/i);
    });
  });

  describe('Default behaviors', function () {
    it('should have default value', () => {
      const { input } = renderApp({ ...properties, defaultValue: 'default value' });
      expect(input.value).toBe('default value');
    });
    it('should change value and call onChange event', async function () {
      const handleChange = jest.fn();
      const { input } = renderApp({ ...properties, onChange: handleChange });
      const user = userEvent.setup();

      await user.type(input, 'test');
      expect(input).toHaveValue('test');
      expect(handleChange).toHaveBeenCalledTimes(4);
    });

    // TODO: Remove this.
    // Validations
    // Formatting
  });
});
