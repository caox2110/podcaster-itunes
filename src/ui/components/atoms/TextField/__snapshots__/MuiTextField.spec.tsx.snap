// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MuiTextField Rendering Should render without throwing any errors 1`] = `
<div>
  <div
    aria-autocomplete="none"
    class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
  >
    <label
      class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-12rpily-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="false"
      for="filter-podcast"
      id="filter-podcast-label"
    >
      Search
    </label>
    <div
      class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-uds2cb-MuiInputBase-root-MuiOutlinedInput-root"
    >
      <input
        aria-invalid="false"
        autocomplete="off"
        class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
        data-testid="filter-podcast"
        id="filter-podcast"
        name="filter-podcast"
        placeholder="Filter podcasts"
        type="text"
        value=""
      />
      <fieldset
        aria-hidden="true"
        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="css-1ftyaf0"
        >
          <span>
            Search
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</div>
`;
